src/main/java/uk/gov/pay/connector/charge/service/ChargeService.java

```diff
@@ -761,18 +761,25 @@ private boolean chargeIsInLockedStatus(OperationType operationType, ChargeEntity
 
     private ExternalMetadata storeExtraFieldsInMetaData(TelephoneChargeCreateRequest telephoneChargeRequest) {
         HashMap<String, Object> telephoneJSON = new HashMap<>();
-        telephoneJSON.put("processor_id", telephoneChargeRequest.getProcessorId());
+        telephoneJSON.put("processor_id", telephoneChargeRequest.getProcessorId().substring(0, Math.min(telephoneChargeRequest.getProcessorId().length(), 50)));
         telephoneJSON.put("status", telephoneChargeRequest.getPaymentOutcome().getStatus());
         
-        telephoneChargeRequest.getCreatedDate().ifPresent(createdDate -> telephoneJSON.put("created_date", createdDate));
-        telephoneChargeRequest.getAuthorisedDate().ifPresent(authorisedDate -> telephoneJSON.put("authorised_date", authorisedDate));
-        telephoneChargeRequest.getAuthCode().ifPresent(authCode -> telephoneJSON.put("auth_code", authCode));
-        telephoneChargeRequest.getTelephoneNumber().ifPresent(telephoneNumber -> telephoneJSON.put("telephone_number", telephoneNumber));
-        telephoneChargeRequest.getPaymentOutcome().getCode().ifPresent(code -> telephoneJSON.put("code", code));
+        telephoneChargeRequest.getCreatedDate().ifPresent(createdDate -> telephoneJSON.put("created_date", createdDate.substring(0, Math.min(createdDate.length(), 50))));
+        telephoneChargeRequest.getAuthorisedDate().ifPresent(authorisedDate -> telephoneJSON.put("authorised_date", authorisedDate.substring(0, Math.min(authorisedDate.length(), 50))));
+        telephoneChargeRequest.getAuthCode().ifPresent(authCode -> telephoneJSON.put("auth_code", authCode.substring(0, Math.min(authCode.length(), 50))));
+        telephoneChargeRequest.getTelephoneNumber().ifPresent(telephoneNumber -> telephoneJSON.put("telephone_number", telephoneNumber.substring(0, Math.min(telephoneNumber.length(), 50))));
```

Comment:
Can we also log message when truncating `authCode`, `telephoneNumber` ? May be extract into small method and use for all truncated fields

```suggestion
       checkAndGetTruncatedValue("telephone_number", telephoneNumber);
```

```
private String checkAndGetTruncatedValue(String field, String value) {
        if (value.length() > 50) {
            logger.info(String.format("Telephone payment message - [%s] is longer than 50 characters and has been truncated and stored. Actual value is [%s]", field, value));
            return value.substring(0, 50);
        }
        return value;
    }
```

Reply:
Done